pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,207 @@
+#include "lists.h"
+
+int tSizes[8] = {
+            1,
+            sizeof(int8_t), 
+            sizeof(int16_t), 
+            sizeof(int32_t), 
+            sizeof(int64_t), 
+            sizeof(float), 
+            sizeof(double), 
+            sizeof(void *)
+};
+
+t_list   *allocate_list(size_t size, t_type type, void *elemnts) // va_arg could alow us to take in multiple list of elements
+{
+    t_list *list;
+
+    if (!(list = (t_list*)malloc(sizeof(t_list))))
+        return NULL;
+    list->size = size * tSizes[type];
+    list->type = type;
+    list->capacity = DEFAULT_CAP;
+    while (list->capacity <= list->size)
+        list->capacity *= 2;
+    if (list->size == 0)
+        list->elements = NULL;
+    else if (!(list->elements = malloc(list->capacity * tSizes[type])))
+        return NULL;
+    if (elemnts)
+        memcpy(list->elements, elemnts, tSizes[type] * list->size);
+    return list;
+}
+
+void     free_list(t_list **list)
+{
+    free((*list)->elements);
+    free(*list);
+    *list = NULL;
+}
+
+void    append(t_list** list1, t_list* list2)
```

Comment:
@azouiten We can definitely make use of the function `insert` here as it would make this function much simpler.

Reply:
I think the second argument `t_list* list2` should be a `void *` instead of `t_list *` because we can append not only `t_list` type.

Reply:
I was thinking we could skip this function altogether as we could simply call insert in the printing.